src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java

```diff
@@ -1239,13 +1239,37 @@ public void visitSources(SCMSourceObserver observer) throws IOException, Interru
         }
     }
 
-    private Iterable<GHRepository> searchRepositories(final GitHub github, final GitHubSCMNavigatorContext context) {
+    private Iterable<GHRepository> searchRepositories(
+            final GitHub github, final GitHubSCMNavigatorContext context, final GHOrganization org) {
         final GHRepositorySearchBuilder ghRepositorySearchBuilder = github.searchRepositories();
         context.getTopics().forEach(ghRepositorySearchBuilder::topic);
         ghRepositorySearchBuilder.org(getRepoOwner());
         if (!context.isExcludeForkedRepositories()) {
             ghRepositorySearchBuilder.q("fork:true");
         }
+
+        // Only the first 1000 search results are available
+        if (ghRepositorySearchBuilder.list().getTotalCount() > 1000) {
+            return org.listRepositories(100);
```

Comment:
Looks to me like this is ignoring all the search settings above.
There is also redundant code in these two new methods that can be reduced.

Reply:
Hello, the query above only returns the first 1000 records. This is a github limitation.
The query also returns the number of existing records and this value is used to make a decision.